src/sonic-config-engine/minigraph.py

```diff
@@ -118,17 +120,19 @@ def parse_png(png, hname):
                 startport = link.find(str(QName(ns, "StartPort"))).text
                 bandwidth_node = link.find(str(QName(ns, "Bandwidth")))
                 bandwidth = bandwidth_node.text if bandwidth_node is not None else None
-
-                if enddevice.lower() == hname.lower():
+                if (enddevice.lower() == hname.lower()):
                     if port_alias_map.has_key(endport):
                         endport = port_alias_map[endport]
                     neighbors[endport] = {'name': startdevice, 'port': startport}
+                    device_neighbors.append(startdevice)
                     if bandwidth:
                         port_speeds[endport] = bandwidth
-                else:
+
+                if (startdevice.lower() == hname.lower()):
```

Comment:
In case of multi-asic platform, when parsing mingraph to generate global config, the parse_png() will be called. In this case, it can happen that start device or end device is not matching with the device name as it can be asic names. So the else condition might be hit often and capturing this data will not be helpful or is not really an error. "elif" can be added, but else condition is not a good idea here.

Reply:
as you have separated the device png parsing and asic png parsing, why do you still need to change device png parsing?

Reply:
I do not understand the argument here. why not use elif here? for else since it does not match hname in either way, just ignore those.

Reply:
Changed to elif

Reply:
added elif